### PR TITLE
Fixes #4709: When a node is deleted, its generated promises are not removed

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/4709

### CODE REVIEW

File: rudder-core/src/main/resources/hooks.d/node-post-deletion/10-clean-policies [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+# Hooks parameter are passed by environment variable: 
+#
+# - RUDDER_NODE_ID: the nodeId
+# - RUDDER_NODE_HOSTNAME: the node fully qualified hostname
+# - RUDDER_NODE_POLICY_SERVER_ID: the node policy server id
+# - RUDDER_AGENT_TYPE : agent type ("cfengine-nova" or "cfengine-community")
+# - RUDDER_POLICIES_DIRECTORY_CURRENT: the full path of the base directory containing policies for that node
+# - RUDDER_POLICIES_DIRECTORY_NEW: the full path of the base directory containing next policies for that node (during a generation)
+# - RUDDER_POLICIES_DIRECTORY_ARCHIVE: the full path of the base directory containing previous policies for that node
```
Comment (ncharles) [inline rudder-core/src/main/resources/hooks.d/node-post-deletion/10-clean-policies]: we should remove this one also

File: rudder-core/src/main/resources/hooks.d/node-post-deletion/10-clean-policies [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# Errors code on hooks are interpreted as follow:
+# - 0     : success, no log (appart if debug one)          , continue to next hook
+# - 1-31  : error  , error   log in /var/log/rudder/webapp/, stop processing
+# - 32-63 : warning, warning log in /var/log/rudder/webapp/, continue to next hook
+# - 64-255: reserved for futur use case. Behavior may change without notice. 
+
+# delete all generated policies for that node
+NODE_DIR=$(dirname ${RUDDER_POLICIES_DIRECTORY_CURRENT})
+
+if [ -d ${NODE_DIR} ]; then
```
Comment (peckpeck) [inline rudder-core/src/main/resources/hooks.d/node-post-deletion/10-clean-policies]: "${NODE_DIR}"

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

**Pull request merged**